crates/database/src/orders.rs

```diff
@@ -498,6 +506,8 @@ pub struct FullOrderWithQuote {
     pub quote_gas_amount: Option<f64>,
     pub quote_gas_price: Option<f64>,
     pub quote_sell_token_price: Option<f64>,
+    pub quote_verified: Option<bool>,
+    pub quote_metadata: Option<serde_json::Value>,
```

Comment:
Does the database layer need to know what is stored in `metadata` field? I think not in this particular case.
I wanted to separate specific type which orderbook/autopilot want to store in that field from the database layer, because we can easily change that type in the future, according to @MartinquaXD comment we will be storing other values here mainly for the debug/analysis purposes. In such approach we don't need to change any type in database, but only orderbook/autopilot code. That is why json value is enough here. The database ensures that value of `metadata` column is a valid json, and upper layer can serialize/deserialize to specific type.
Alternatively we could use generics with trait bounds like: `T: Into<serde_json::Value>` but we will still use only one type in orderbook/autopilot for the `metadata` field.

Reply:
I think at this point we indeed do not have to have a concrete type yet. To keep complexity low I'd either simply pipe the metadata into the `GET /orders` API response or validate before sending the response.